packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/views/explorer/services/ExpandAllTreePathProvider.java

```diff
@@ -124,4 +129,25 @@ private int addAllContents(IEditingContext editingContext, String treeItemId, in
 
         return depthConsidered;
     }
+
+    private Object getTreeItemObject(IEditingContext editingContext, String treeItemId, Tree tree) {
+        var optionalTreeDescription = this.getTreeDescription(editingContext, tree);
+
+        if (optionalTreeDescription.isPresent()) {
+            var variableManager = new VariableManager();
+            variableManager.put(IEditingContext.EDITING_CONTEXT, editingContext);
+            variableManager.put(TreeDescription.ID, treeItemId);
+            return optionalTreeDescription.get().getTreeItemObjectProvider().apply(variableManager);
+        }
+        return null;
+    }
+
+    private Optional<TreeDescription> getTreeDescription(IEditingContext editingContext, Tree tree) {
+        return this.representationDescriptionSearchService
+        .findAll(editingContext).values().stream()
+        .filter(TreeDescription.class::isInstance)
+        .map(TreeDescription.class::cast)
+        .filter(td -> Objects.equals(td.getId(), tree.getDescriptionId()))
+        .findFirst();
```

Comment:
Can be removed and replaced inline with a simple call to `IRepresentationDescriptionSearchService#findById` instead

Reply:
Done